tests/integration/helpers/css-properties-test.js

```diff
@@ -34,7 +34,12 @@ test('it should escape malicious property values', function(assert) {
     '10px',
     'Passes first set value in potentially malicious values'
   );
-  assert.notEqual(el.style.color, 'red', 'Does not pass values after a ";"');
+
+  assert.equal(
+    el.outerHTML,
+    '<div data-test-malicious="" style="width:10px; color: red;&quot;><h2>Hello World</h2></div><div style=&quot;;"></div>',
+    'Generates safely escaped malicious content'
```

Comment:
This is not safe _at all_. 😅 
If you re-add the assertion you have removed, you can see why.

```js
assert.notEqual(el.style.color, 'red', 'Does not pass values after a ";"');
```

Reply:
That is just mean 🥴

I wouldn’t know how to solve it without some major parsing of braces indeed.

What CSS values exists where a ; is valid other then url()?

Reply:
At least according to Chrome, `()`, `[]` and `{}` constitute expressions that may contain unescaped semicolons:

```
<div style="a: (; b: )[; c: ]{; d: }; e: foo;"></div>
```

![image](https://user-images.githubusercontent.com/834636/49884011-2a865000-fe34-11e8-9a85-1fc20fceb17e.png)

I don't know about other browsers or outdated versions though, so I would like to be _super_ conservative here and only add further exemptions, when new bug reports come flying in.

So for now, I would only allow `;` somewhere inside of `()`. I think we should allow arbitrary nesting of `()` as well.

Since this is a recursive problem, I don't think we can solve this with a regular expression and would need to parse the string manually. I also believe that this is a problem should be handled upstream by `cssesc`.

If you have some spare time, I would appreciate it, if you could open an issue over there and link it here.

Reply:
Yes, it seems to be a concern for ‘cssecs’. Ill create an issue there. Thanks